src/data/value/into.rs

```diff
@@ -0,0 +1,107 @@
+use {
+    super::{Value, ValueError},
+    crate::result::{Error, Result},
+    std::convert::TryInto,
+};
+
+macro_rules! impl_into {
+    ($type:ident, $($matcher:pat => $result:expr),+) => {
+        impl Into<$type> for Value {
+            fn into(self) -> $type {
+                (&self).into()
+            }
+        }
+        impl Into<$type> for &Value {
+            fn into(self) -> $type {
+                match self {
+                    $($matcher => $result),+
+                }
+            }
+        }
+    };
+}
+
+macro_rules! impl_try_into {
+    ($type:ident, $($matcher:pat => $result:expr),+) => {
+        impl TryInto<$type> for Value {
+            type Error = Error;
+            fn try_into(self) -> Result<$type> {
+                (&self).try_into()
+            }
+        }
```

Comment:
Though this might be convenient for making users not to think about their codes is using whether `Value` or `&Value`, I also concern that this also can make users confused by providing two different ways to do a same task.

In the case of `impl Into<String> for Value` and `impl into<String> for &Value`, if user can use `Value` then it is always better to use `impl Into<String> for Value`.

However, these `TryInto` cases are different, both are identical.
If `impl TryInto<$type> for Value` can do anything better than `impl Try<$type> for &Value`, then it can be worth enough to have both. However, it is not in this case.

Let's have only `impl TryInto<$type> for &Value`, and then `impl_try_into`  macro is also unnecessary.
How do you think?

Reply:
See https://github.com/gluesql/gluesql/pull/187#discussion_r601050195